core-libs/setup/ssr/optimized-engine/request-context.ts

```diff
@@ -9,44 +9,73 @@ import { Request } from 'express';
 import { ExpressServerLogger, ExpressServerLoggerContext } from '../logger';
 import { parseTraceparent } from '../logger/loggers/w3c-trace-context/parse-traceparent';
 
+const getRequestContext = (request: Request) => {
+  return request.res.locals.cx.request;
+};
+
+const setRequestContext = (
+  request: Request,
+  context: ExpressServerLoggerContext
+) => {
+  request.res.locals = {
+    cx: {
+      request: context,
+    },
+  };
+};
```

Comment:
Let's not overwrite the whole `locals` object. Other features might also contributes to it. Same holds for the `cx` object - other features might contribute to it.

Let's do safe spreading on each level.